x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/accounts_stats_collector.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { calculatePostureScore } from '../../../routes/compliance_dashboard/get_stats';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+import type { CspmAccountsStats } from './types';
+
+// TODO: test with multiple accounts
+// TODO: naming, naming, naming
+
+export interface Accounts {
+  accounts: {
+    buckets: AccountEntity[];
+  };
+}
+
+export interface AccountEntity {
+  key: string; // account_id
+  doc_count: number;
+  resource_type: {
+    buckets: ResourceType[];
+  };
+}
+
+export interface ResourceType {
+  key: string;
+  doc_count: number;
+  evaluation: { buckets: StatsEntity[] };
+}
+
+export interface StatsEntity {
+  key: string;
+  doc_count: number;
+  agents: Agents;
+}
+export interface Agents {
+  value: number;
+}
+
+export interface ResourceStats {
+  [key: string]: {
+    doc_count: number;
+    passed: number;
+    failed: number;
+  };
+}
+
+export const getAccountsStats = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<CspmAccountsStats[]> => {
+  try {
+    const isIndexExists = await esClient.indices.exists({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+    });
+
+    if (isIndexExists) {
+      return await getAccountStatsByResourceType(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS);
+    }
+
+    return [];
+  } catch (e) {
+    logger.error(`Failed to get accounts stats ${e}`);
+    return [];
+  }
+};
+
+const getClusterScore = (evaluationByResourceType: ResourceStats): number => {
+  let passed = 0;
+  let failed = 0;
+  Object.entries(evaluationByResourceType).forEach(([key, value], index) => {
```

Comment:
if `key` is not used, consider `Object.values()` instead

_nit_

i think (not sure by interface) that we're adding `0` every iteration to the key we don't have.